src/com/googlecode/jmxtrans/model/output/GangliaWriter.java

```diff
@@ -62,6 +64,7 @@
 	public static final int DEFAULT_DMAX = 0;
 	public static final int DEFAULT_TMAX = 60;
 	public static final String DEFAULT_GROUP_NAME = "JMX";
+	public static final BigDecimal GANGLIA_MIN_DOUBLE = new BigDecimal("1E-308");
```

Comment:
`1E-308` is actually > 0. Which means that with the test you are doing below, any negative value will be modified to 0. This is probably not what you want to do.